internal/certs/certs.go

```diff
@@ -140,7 +146,7 @@ func WithName(name string) Option {
 
 // New is the main helper to create a certificate, it is recommended to
 // use the more specific ones for specific use cases.
-func New(isCA bool, issuer *Issuer, opts ...Option) (*Certificate, error) {
+func New(isCA, isClient bool, issuer *Issuer, opts ...Option) (*Certificate, error) {
```

Comment:
Nit. Maybe we can make the certificate type some kind of enum, as the options here seem to be mutually exclusive. But we can leave this refactor for later.
```suggestion
const (
     IssueCertTypeCA = iota
     IssueCertTypeService
     IssueCertTypeClient
)
....
func New(certType int, issuer *Issuer, opts ...Option) (*Certificate, error) {
```

Reply:
I can take this up separately. Yeah.